src/v2i-hub/PedestrianPlugin/src/FLIRWebSockAsyncClnSession.cpp

```diff
@@ -255,46 +294,61 @@ namespace PedestrianPlugin
                         }
 
                         PLOG(logINFO) << "Received FLIR camera data at: " << dateTimeArr[0] << "/" << dateTimeArr[1] << "/" << dateTimeArr[2]
-                        << " " << dateTimeArr[3] << ":" << dateTimeArr[4] << ":" << dateTimeArr[5] << ":" << dateTimeArr[6] << std::endl;  
+                        << " " << dateTimeArr[3] << ":" << dateTimeArr[4] << ":" << dateTimeArr[5] << ":" << dateTimeArr[6];  
 
                         PLOG(logINFO) << "Received FLIR camera data for pedestrian " << idResult << " at location: (" << lat << ", " << lon <<
-                        ")" << ", travelling at speed: " << speed << ", with heading: " << alpha << " degrees" << std::endl;
+                        ")" << ", travelling at speed: " << speed << ", with heading: " << alpha << " degrees";
 
-                        PLOG(logINFO) << "PSM message count: " << msgCount << std::endl;
+                        PLOG(logINFO) << "Message count: " << msgCount;
 
-                        //constructing xml to send to BroadcastPSM function
+                        /* NOTE: Removed PSM constuction from FLIR input. Keeping as reference and if returned use is desired.
```

Comment:
Can't you just use a configurable parameter to select what type of message you want to send. You can make the configurable parameter a comma separate list of approved types. Something like SDSM,TIM would trigger you to send and SDMS and TIM while something like PSM would just send a PSM

Reply:
Added configurable parameter that accepts SDSM, TIM, PSM in any combination